learning-bitmark/contributing-to-bitmark/bitmark-software-development-process.md

```diff
@@ -0,0 +1,127 @@
+
+![](https://github.com/bitmark-inc/docs/blob/master/learning-bitmark/contributing-to-bitmark/imgs/project_bitmarkd.png)
+
+
+**Branch, Tag and release process:**
+
+We have a single development branch master, from where we create tags to perform minor and major releases. The development never happens directly on branch master, instead the developer create a development branch from master, after the developer believes its branch is ready, it opens a pull request and assign another developer to make a review. After the review process, the developer merges its branch to master branch and deletes the development branch. The developers never commit directly into the master branch without passing through the review process.
+
+![](https://github.com/bitmark-inc/docs/blob/master/learning-bitmark/contributing-to-bitmark/imgs/github_branches.png)
+
+
+**Review process:**
+
+We have a peer review process, where the developer can assign one or more individuals to verify its code, these individuals shall suggest modifications or accept the pull request, also the reviewer can reassign the review. Currently there are no time outs specified for reviewers, and in such a case, if a reviewer is delaying the review, it must discuss during the stand-up daily meeting.
+
+![](https://github.com/bitmark-inc/docs/blob/master/learning-bitmark/contributing-to-bitmark/imgs/review-process.png)
+
+# Development work-flow improvements
+
+**Development process:**
+
+This is our working flow using GitHub. Here I describe 3 basic areas of our development process.
+
+
+![](https://github.com/bitmark-inc/docs/blob/master/learning-bitmark/contributing-to-bitmark/imgs/new-review-process.png)
+
+
+**What to do when there is a new patch?**
+
+First, we need to ask ourselves, is it a trivial patch? In case it is not a 3-lines change patch, it means, it is not a trivial patch, sometimes even a 3-lines change is not trivial. We do need to be mature enough to understand what is and what is not a trivial patch. In case it is not a trivial patch, we shall open a GitHub issue with a proper title and describe what is the problem we are trying to fix, following with a pull request with a patch (if we have one), this patch must point out to the issue number. All the discussion related to implementation, shall be in the issue, code problems shall be in the pull request.
+
+**I would like to stress a bit how to perform a proper code review:**
+
+We need to always remember, there are other people watching us, the information is on the Internet and it will always be there, we need developers to engage with us, perhaps we need companies out there to see us relevant and join our cause.
+
+**What to do:**
+
```

Comment:
A general suggestion for lists of this type: keep all the items on a list in the same style. Often, that means to start them with an imperative (command) word. In this case, lines 1, 2, 4, 5, 6, 8, and 10 all follow this formula, which makes the others stand out more. #3 (if it's appropriate to include) would be something like "Determine if it passes ...", #7 could be "Remember that Peer review ..." etc.